Libraries/Utilities/warnMoved.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+const warning = require('fbjs/lib/warning');
+
+const warnedApis: {[string]: boolean} = {};
+
+/**
+ * A simple function that warns the user once if an API has been moved to
+ * another module.
+ *
+ * @param {string} api - The name of the API or component that moved
+ * @param {string} destModule - The name of the module that the API moved to
+ * @param {string} [renamed] - Optional; name of the API in the dest module,
+ *                             if it was renamed
+ */
+function warnMoved(api: string, destModule: string, renamed?: string) {
```

Comment:
If you make another function called warnOnce then it can be used for the existing warnings in the api as well.

Reply:
sounds like a good idea! with the delay, maybe I should instead split this helper function into its own pull request?

Reply:
Split maybe a good idea. We may have more usage for this if we move more component out.